Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,610 @@
+            return nil
+        }
+        if signatures[.objectiveC]?.returns == [.init(kind: .typeIdentifier, spelling: "BOOL", preciseIdentifier: "c:@T@BOOL")] {
+            // If the Objective-C function signature returns BOOL, don't add anything
+            return nil
+        }
+        
+        let lastSentenceEndsWithPunctuation = returns.contents.last?.format().removingTrailingWhitespace().last?.isPunctuation == true
+        if let inlineContents = returns.contents as? [InlineMarkup] {
+            return [Paragraph(inlineContents + objcObjectErrorAddition(endPreviousSentence: !lastSentenceEndsWithPunctuation))]
+        } else if let paragraphs = returns.contents as? [Paragraph] {
+            guard let lastParagraph = paragraphs.last else {
+                return [Paragraph(objcObjectErrorAddition(endPreviousSentence: false))]
+            }
+            return paragraphs.dropLast() + [Paragraph(lastParagraph.inlineChildren + objcObjectErrorAddition(endPreviousSentence: !lastSentenceEndsWithPunctuation))]
+        } else {
+            return returns.contents + [Paragraph(objcObjectErrorAddition(endPreviousSentence: false))]
+        }
+    }
+    
+    /// The known declaration fragment alternatives that represents "void" in each programming language.
+    private static var knownVoidReturnValuesByLanguage: [SourceLanguage: [SymbolGraph.Symbol.DeclarationFragments.Fragment]] = [
+        .swift: [
+            // The Swift symbol graph extractor uses one of these values depending on if the return value is explicitly defined or not.
+            .init(kind: .text, spelling: "()", preciseIdentifier: nil),
+            .init(kind: .typeIdentifier, spelling: "Void", preciseIdentifier: "s:s4Voida"),
+        ],
+        .objectiveC: [
+            .init(kind: .typeIdentifier, spelling: "void", preciseIdentifier: "c:v"),
+        ]
+    ]
+    
+    /// Translates a relative documentation comment source range to the absolute location in that source file.
+    private func adjusted(_ range: SourceRange?) -> SourceRange? {
+        range.map { adjusted($0) }
+    }
+    
+    /// Translates a relative documentation comment source range to the absolute location in that source file.
+    private func adjusted(_ range: SourceRange) -> SourceRange {
+        var range = range
+        if let source = range.source {
+            for case let .sourceCode(location?, offset?) in docChunkSources where location.url == source {
+                range.offsetWithRange(offset)
+                return range
+            }
+        }
+        return range
+    }
+    
+    // MARK: Diagnostics
+    
+    /// Creates a new problem about return value documentation for a symbol that returns void.
+    ///
+    /// ## Example
+    ///
+    /// ```swift
+    /// /// - Paramaters:
+    /// ///   - someValue: Some description of this parameter
+    /// /// - Returns: Description of what this function returns
+    /// ///   ^~~~~~~
```

Comment:
I love these examples of the error messages and how they will appear - thank you!